src/middleware/authorisation.coffee

```diff
@@ -19,134 +15,57 @@ os = require 'os'
-    return matchJsonPath channel.matchContentJson, channel.matchContentValue, body
-  else if channel.matchContentXpath or channel.matchContentJson
-    # if only the match expression is given, deny access
-    # this is an invalid channel
-    logger.error 'Channel with name "' + channel.name + '" is invalid as it has a content match expression but no value to match'
-    return false
-  else
-    return true
-
-matchRegex = (regexPat, body) ->
-  regex = new RegExp regexPat
-  return regex.test body.toString()
-
-matchXpath = (xpathStr, val, xml) ->
-  doc = new dom().parseFromString(xml.toString())
-  xpathVal = xpath.select(xpathStr, doc).toString()
-  return val == xpathVal
-
-matchJsonPath = (jsonPath, val, json) ->
-  jsonObj = JSON.parse json.toString()
-  jsonVal = getJSONValByString jsonObj, jsonPath
-  return val == jsonVal.toString()
-
-# taken from http://stackoverflow.com/a/6491621/588776
-# readbility improved from the stackoverflow answer
-getJSONValByString = (jsonObj, jsonPath) ->
-  jsonPath = jsonPath.replace(/\[(\w+)\]/g, '.$1')  # convert indexes to properties
-  jsonPath = jsonPath.replace(/^\./, '')            # strip a leading dot
-  parts = jsonPath.split('.')
-  while parts.length
-    part = parts.shift()
-    if part of jsonObj
-      jsonObj = jsonObj[part]
-    else
-      return
-  return jsonObj
-
-extractContentType = (ctHeader) ->
-  index = ctHeader.indexOf ';'
-  if index isnt -1
-    return ctHeader.substring(0, index).trim()
-  else
-    return ctHeader.trim()
-
 genAuthAudit = (remoteAddress) ->
   audit = atna.nodeAuthentication remoteAddress, himSourceID, os.hostname(), atna.OUTCOME_MINOR_FAILURE
   audit = atna.wrapInSyslog audit
   return audit
 
+authoriseClient = (channel, ctx) ->
+  if ctx.authenticated? and channel.allow?
+    if ctx.authenticated.roles?
+      match = false
+      channel.allow.forEach (role) ->
+        if ((ctx.authenticated.roles.indexOf role) isnt -1)
+          match = true
+          return
+      if match
+        return true
+    if ((channel.allow.indexOf ctx.authenticated.clientID) isnt -1)
+      return true
+
+  return false
+
+authoriseIP = (channel, ctx) ->
+  if channel.whitelist?.length > 0
+    return (channel.whitelist.indexOf ctx.ip) isnt -1
+  else
+    return false
 
-# Is the channel enabled?
-# If there is no status field then the channel IS enabled
-exports.isChannelEnabled = isChannelEnabled = (channel) -> not channel.status or channel.status is 'enabled'
-
-channelWithHighestPriority = (channels) ->
-  channels.reduce (channel1, channel2) ->
-    # let channel1 have precedence if equal
-    if not channel2.priority? then return channel1
-    if not channel1.priority? then return channel2
-    if channel1.priority <= channel2.priority then return channel1
-    return channel2
+authFunctions = [
+  authoriseClient,
+  authoriseIP
+]
 
+isAuthorised = (channel, ctx) ->
+  return authFunctions.some (authFunc) ->
```

Comment:
hmmm i don't think this is correct - so in this case _anyone_ who has a whitelisted IP is authenticated?

The whitelisting, if enabled, should happen _in addition_ to the authoriseClient check